domains/Discussions/Tests/Feature/QuestionsIndexTest.php

```diff
@@ -0,0 +1,218 @@
+    {
+        $this->actingAs($this->user);
+
+        for ($attempt = 0; $attempt < 30; ++$attempt) {
+            $this->get(route('discussions.questions.index'));
+        }
+
+        $this->get(route('discussions.questions.index'))
+            ->assertResponseStatus(Response::HTTP_OK);
+    }
+
+    /** @test */
+    public function it_navigates_to_next_page(): void
+    {
+        $this->json('GET', route('discussions.questions.index', [
+            'page' => 2,
+        ]))
+            ->seeJsonContains([
+                'current_page' => 2,
+            ]);
+    }
+
+    /** @test */
+    public function it_search_by_author(): void
+    {
+        $user = UserFactory::new()->create();
+        QuestionFactory::new([
+            'author_id' => $user->id,
+        ])
+            ->count(3)
+            ->create();
+
+        $this->json('GET', route('discussions.questions.index', [
+            'author' => $user->id,
+        ]))
+            ->seeJsonContains([
+                'email' => $user->email,
+            ])
+            ->seeJsonContains([
+                'to' => 3,
+            ]);
+    }
+
+    /** @test */
+    public function it_search_by_title(): void
+    {
+        QuestionFactory::new([
+            'title' => 'LARAVEL-PT',
+        ])->create();
+        QuestionFactory::new([
+            'title' => 'laravel-Pt',
+        ])
+            ->create();
+
+        $this->json('GET', route('discussions.questions.index', [
+            'title' => 'LArAvEL-pT',
+        ]))
+            ->seeJsonContains([
+                'to' => 2,
+            ]);
+    }
+
+    /** @test */
+    public function it_search_by_created_date(): void
+    {
+        QuestionFactory::new([
+            'created_at' => Carbon::now()->subYears(2),
+        ])
+            ->create();
+        QuestionFactory::new([
+            'created_at' => Carbon::now()->subYears(3),
+        ])
+            ->create();
+
+        $this->json('GET', route('discussions.questions.index', [
+            'created[from]' => Carbon::now()->subMonth()->subYears(2)->toDateString(),
+            'created[to]' => Carbon::now()->addMonth()->subYears(2)->toDateString()
+        ]))
+            ->seeJsonContains([
+                'to' => 1,
+            ]);
+
+        $this->json('GET', route('discussions.questions.index', [
+            'created[from]' => Carbon::now()->subMonth()->subYears(3)->toDateString(),
+            'created[to]' => Carbon::now()->addMonth()->subYears(2)->toDateString()
+        ]))
+            ->seeJsonContains([
+                'to' => 2,
+            ]);
+    }
+
+    /** @test */
+    public function it_search_by_resolved(): void
+    {
+        QuestionFactory::new([
+            'resolved_at' => Carbon::now(),
+        ])
+            ->count(5)
+            ->create();
+
+        $this->json('GET', route('discussions.questions.index'))
+            ->seeJsonContains([
+                'to' => 15,
+            ]);
```

Comment:
I think this assertion isn't necessary for this test. There's already a test to test it lists all non-deleted links.

Reply:
Yes, I forget to remove these lines.

Reply:
@tiagof, I added five resolved questions, I'm testing to verify that when a resolved param is not sent the questions are all returned.

Reply:
I understand, but it's not what this test case is testing. We added 5 resolved questions and when we search for them, those 5 questions are returned. 

The fact that they return 15 when not having a search param is subject to change (e.g..: if the number of questions to show changes in the future) and that not only will require to change the directly-related test case but also this one. Test cases should be as small as possible in terms of scope, for that reason.